### ISSUE TITLE
password protected pdf validation

### ISSUE THREAD

**Issue by chiragmewada** [pqina/filepond#176]
Hello fellow developer!

Thanks for the state of the art file upload library.

I am using filepond in my application and it is implemented using jQuery. I have a unique request which I think if incorporated in the library will be great addons.

In my application user is uploading pdf files, in most cases, the user upload **password protected pdf** file.

I am able to validate it on the server side. Is there a way I can validate same on the client side so that I can show an appropriate message to my user and stopping them to upload password protected pdf files.

Thanks,
Chirag.

**Comment by rikschennink** [comment]
I've just published version `3.5.0` of FilePond which now contains a `beforeAddFile` hook. You can use this to do custom file validation.

See the list below for a couple of StackOverflow answers on determining if a PDF is password protected in the browser:
- https://stackoverflow.com/questions/44731359/detect-in-browser-if-pdf-is-locked-or-encrypted
- https://stackoverflow.com/questions/52054746/read-pdf-file-with-html5-file-api

```js
var pond = FilePond.create(el, {
    allowMultiple: true,
    instantUpload: false,
    beforeAddFile: (fileItem) => new Promise(resolve => {
        // Read PDF content here with FileReader or PDF.js
       

        // when PDF is not password protected run
        return resolve(true);

        // when PDF is password protected run
        return resolve(false);
    });
});
```

**Comment by chiragmewada** [comment]
Thanks a lot for addressing my request.

Best Regards,
Chirag.
